vm/src/api/de.rs

```diff
@@ -856,6 +856,38 @@ impl<'de, 'a, 't> VariantAccess<'de> for Enum<'a, 'de, 't> {
     where
         V: Visitor<'de>,
     {
-        de::Deserializer::deserialize_seq(self.de, visitor)
+        let typ = resolve::remove_aliases_cow(self.de.state.env, &mut NullInterner, self.de.typ);
+        match (self.de.input.as_ref(), &**typ) {
+            (ValueRef::Data(data), &Type::Variant(ref row)) => {
```

Comment:
Quite a lot of rightward drift here, would be easier to read if it is rewritten in this style

```rust
let (data, row) = match (self.de.input.as_ref(), &**typ) {
    (ValueRef::Data(data), &Type::Variant(ref row)) => (data, row),
    _=> return Err(..)
}
```